frontend/src/components/navbar/ProductSearch.tsx

```diff
@@ -0,0 +1,25 @@
+import { useLocation } from 'react-router-dom';
+import SearchIcon from "@mui/icons-material/Search";
+
+import { SearchContainer, SearchIconWrapper, SearchInput } from "./MuiComponents";
+import { HOME_PATH } from '../constants';
+
+const ProductSearch = () => {
+
+    const location = useLocation();
+    const isHomePage = location.pathname === HOME_PATH;
```

Comment:
A better pattern for using `useLocation` without repeating the logic in every component is to create a custom hook that encapsulates the functionality and can be reused across multiple components